rust_src/src/numbers.rs

```diff
@@ -158,6 +159,27 @@ impl IsLispNatnum for EmacsInt {
     }
 }
 
+pub trait CheckRange {
+    fn check_range(self, from: EmacsInt, to: EmacsInt);
+}
+
+impl CheckRange for EmacsInt {
```

Comment:
Instead of making this a trait, why not have it be a simple function?

```
pub fn check_range(num: impl Into<EmacsInt>, from: impl Into<EmacsInt>, to: impl Into<EmacsInt>) {
    let num: EmacsInt = num.into();
    let from: EmacsInt = from.into();
    let to: EmacsInt = to.into();
    if !(from <= self && self <= to) {
        args_out_of_range!(
            self,
            if from < 0 && from < MOST_NEGATIVE_FIXNUM {
                MOST_NEGATIVE_FIXNUM
            } else {
                from
            },
            cmp::min(to, MOST_POSITIVE_FIXNUM)
        )
    }
}
```

Since the arguments are all generic, you do not need to call `into` everywhere. Would any type implement this function differently?

Reply:
That's definitely the nicest way of doing it. Thanks for the pointer.

Reply:
Hat tip to @nickdrozd who has been spear heading this lately.